src/python/include/Delaunay_complex_interface.h

```diff
@@ -0,0 +1,108 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2016 Inria
+ *
+ *    Modification(s):
+ *      - 2024/03 Vincent Rouvreau: Renamed Alpha_complex_interface as Delaunay_complex_interface for DelaunayCechComplex.
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef INCLUDE_DELAUNAY_COMPLEX_INTERFACE_H_
+#define INCLUDE_DELAUNAY_COMPLEX_INTERFACE_H_
+
+#include "Delaunay_complex_factory.h"
+#include <gudhi/Alpha_complex_options.h>
+#include <gudhi/MEB_filtration.h>
+
+#include "Simplex_tree_interface.h"
+
+#include <CGAL/Epeck_d.h>
+#include <CGAL/Epick_d.h>
+
+#include <vector>
+#include <memory>  // for std::unique_ptr
+#include <cstddef>  // for std::size_t
+
+namespace Gudhi {
+
+namespace delaunay_complex {
+
+class Delaunay_complex_interface {
+ public:
+  Delaunay_complex_interface(const std::vector<std::vector<double>>& points,
+                             const std::vector<double>& weights,
+                             bool fast_version, bool exact_version) {
+    const bool weighted = (weights.size() > 0);
+    // Specific cases for dimensions 2 and 3
+    const std::size_t dimension = ((points.size() > 0) ? points[0].size() : 0);
+    if (fast_version) {
+      if (weighted) {
+        if (dimension == 2) {
```

Comment:
It looks like even CircleCI fails to compile all those versions in time, we may have to split the instantiations into several translation units (I think we can write `sources=['file.pyx','other.cpp']` in setup.py).
We could also divide by 2 the number of triangulations, since Epick_d and Epeck_d build the same combinatorial structure, but that may require major design changes.

Reply:
Let's try to modify the [resource-class](https://circleci.com/docs/configuration-reference/#resourceclass) for circle CI. On my laptop, it requires 5 Go of RAM.

Reply:
Ok with large on [b5d186f](https://github.com/GUDHI/gudhi-devel/pull/1052/commits/b5d186ff7f16169711fccf1bda5ca6d1f77ec593) (20 credits per minute instead of 10 - free plan is 400.000 credits per year for circle CI)

Reply:
Strange, usually the limiting factor was the windows CI, but it passes :confused: 
We probably won't be able to avoid splitting forever though.